internal/controllers/core/filewatch/controller.go

```diff
@@ -15,31 +15,93 @@ package filewatch
 import (
 	"context"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
 	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 
-	"github.com/tilt-dev/tilt/internal/store"
 	filewatches "github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
+	"github.com/tilt-dev/tilt/pkg/logger"
 )
 
-// FileWatchController reconciles a FileWatch object
+// fsFileWatchFinalizer cleans up the actual filesystem (fs) monitor for a FileWatch spec.
+const fsFileWatchFinalizer = "fswatch.filewatch.finalizers.core.tilt.dev"
+
+// Controller reconciles a filewatches.FileWatch object.
 type Controller struct {
 	ctrlclient.Client
-	Store store.RStore
+
+	WatchManager *ApiServerWatchManager
 }
 
-func NewController(store store.RStore) *Controller {
+func NewController(wm *ApiServerWatchManager) *Controller {
 	return &Controller{
-		Store: store,
+		WatchManager: wm,
 	}
 }
 
 // +kubebuilder:rbac:groups=,resources=filewatches,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=,resources=filewatches/status,verbs=get;update;patch
 // +kubebuilder:rbac:groups=,resources=filewatches/finalizers,verbs=update
 
-func (r *Controller) Reconcile(_ context.Context, _ ctrl.Request) (ctrl.Result, error) {
-	// this is currently a no-op stub
+func (r *Controller) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := logger.Get(ctx).WithFields(logger.Fields{"apiserver_entity": req.NamespacedName.String()})
```

Comment:
fwiw, I've been putting the manifest name as a label on the object, and using that to construct a logger that goes through the normal tilt logging flow...don't want to reinvent everything at once...